### PR TITLE
[TF-1361] Update Web Tooling Usage

### PR DESCRIPTION
**Pull request by cartmouse**
A few of the examples were not using the new `TouchFree.Init()` API to do the initialisation. Those have been updated and some refactoring has been caried out.

### CODE REVIEW

File: Examples-Web/HoverButton/BasicHoverButton.html [hunk 1]
```diff
@@ -1,15 +1,42 @@
...
+<head>
+    <meta charset="UTF-8" />
+    <title>Basic Hover Button Example</title>
+    <script name="TouchFree_Tooling" type="text/javascript" src="./TouchFree/TouchFree_Tooling.js"></script>
+    <link rel="stylesheet" href="./BasicHoverButton.css" />
+</head>
+
+<body>
+    <button class="button" style="background-color: #fdac53" onpointerenter="onHover(this)"
+        onpointerleave="onUnhover(this)" onpointerdown="onDown(this)" onpointerup="onUp(this)">
```
Comment (JordanBerni-UL) [inline Examples-Web/HoverButton/BasicHoverButton.html:13]: 👀 If these were arrow functions then we don't need to bind (this)

Comment (cartmouse) [reply Examples-Web/HoverButton/BasicHoverButton.html:13]: Turns out you still do! We think this is something we're confusing with React behaviour.

File: Examples-Web/AnimatedProgressButtons/CircularProgress/CircularProgressButtons.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        <button class="button" style="background-color:var(--marigold);"></button>
+        <button class="button" style="background-color:var(--illuminating);"></button>
+    </div>
+    <div class="container">
+        <button class="button" style="background-color:var(--mint);"></button>
+        <button class="button" style="background-color:var(--green-ash);"></button>
+        <button class="button" style="background-color:var(--cerulean);"></button>
+        <button class="button" style="background-color:var(--amythist-orchid);"></button>
+    </div>
+    <script type="text/javascript" src="../Hover.js"></script>
```
Comment (JordanBerni-UL) [inline Examples-Web/AnimatedProgressButtons/CircularProgress/CircularProgressButtons.html]: 🤷 Could this be with the other script tags? Or above the divs if you want it scope to <body> (same for the LinearProgress html)

File: Examples-Web/AnimatedProgressButtons/LinearProgress/LinearProgress.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+// Animates a clip path for active overlay elements, identified by the "hovered" id
+function AnimateLinearProgress(inputAction) {
+    const hoveredElement = document.querySelector(".hovered");
+    if (hoveredElement) {
+        const progressPercent = inputAction.ProgressToClick * 100;
+        document.querySelector(":root").style.setProperty("--progress", `${progressPercent}%`);
+    }
+}
+
+function AddTouchFreeCursor() {
```
Comment (JordanBerni-UL) [inline Examples-Web/AnimatedProgressButtons/LinearProgress/LinearProgress.js:16]: 💬 Do we want to use the new Init cursor option? Or leave this as an example of how to add a custom cursor?
(I think we should have specific cursor examples in the future)

File: Examples-Web/IdleTimeout/src/Hooks/IdleTimeout.tsx [hunk 4]
```diff
@@ -1,36 +1,36 @@
...
-const useIdleTimeout = (timeoutMs: number): AppState => {
-    const [appState, setAppState] = React.useState<AppState>('Idle');
+// TouchFree.Init() must be called before this hook.
+const useIdleTimeout = (timeoutMs: number): boolean => {
+    const [isIdle, setIsIdle] = useState(true);
 
     useEffect(() => {
         let timeoutId: NodeJS.Timeout;
-        const timeoutFunction = () => setAppState('Idle');
+        const timeoutFunction = () => setIsIdle(true);
```
Comment (JordanBerni-UL) [inline Examples-Web/IdleTimeout/src/Hooks/IdleTimeout.tsx]: 🤷 This is only used once so don't think it needs it own function

**Review by JordanBerni-UL — COMMENTED** [review]
Working nicely - the new layout is a lot nicer! 

This PR also covers https://ultrahaptics.atlassian.net/browse/TF-1271

File: Examples-Web/AnimatedProgressButtons/LinearProgress/LinearProgress.js (hunk 3 above)
Comment (cartmouse) [reply Examples-Web/AnimatedProgressButtons/LinearProgress/LinearProgress.js:16]: I think this is probably best left as it's a good cursor for this example (has no ring and doesn't resize) leaving the gradient to show progress by itself. I agree that we should create some specific cursor examples though :+1:

File: Examples-Web/IdleTimeout/src/Hooks/IdleTimeout.tsx (hunk 4 above)
Comment (cartmouse) [reply Examples-Web/IdleTimeout/src/Hooks/IdleTimeout.tsx]: Removed!

File: Examples-Web/AnimatedProgressButtons/CircularProgress/CircularProgressButtons.html (hunk 2 above)
Comment (cartmouse) [reply Examples-Web/AnimatedProgressButtons/CircularProgress/CircularProgressButtons.html]: I've made it so this can sit in the `<head>` now

**Review by JordanBerni-UL — APPROVED** [review]

**Pull request merged**